libptable/ptable.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Partition table
+ *
+ * Copyright 2020, 2023 Phoenix Systems
+ * Author: Hubert Buczynski, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _LIB_PTABLE_H_
+#define _LIB_PTABLE_H_
+
+#include <stdint.h>
+
+
+/* Structure of partition table
+ *  _________________________________________________________________________
+ * |       28 B      |                 32 B * n                |     4 B     |
+ * |-----------------|-----------------------------------------|-------------|
+ * | ptable_t header | ptable_part_t 0 | ... | ptable_part_t n | magic bytes |
+ *  -------------------------------------------------------------------------
+ * 
+ *  NOTE: data in partition table should be stored in little endian
+ */
+
+
+/* Partition table magic signature */
+static const uint8_t ptable_magic[] = { 0xde, 0xad, 0xfc, 0xbe };
+
+
+/* Supported partition types */
+enum { ptable_raw = 0x51, ptable_jffs2 = 0x72, ptable_meterfs = 0x75 };
+
+
+typedef struct {
+	uint8_t name[8]; /* Partition name */
+	uint32_t offset; /* Partition offset (in bytes) */
+	uint32_t size;   /* Partition size (in bytes) */
+	uint8_t type;    /* Partition type */
+
+	/* Reserved bytes */
+	union {
+		struct {
+			uint8_t pad[11];
+			uint32_t crc; /* Partition checksum */
+		} __attribute__((packed));
```

Comment:
Nameless structs/unions within structs are GNU extention

Reply:
... or a part of C11 standard (just pointing out for completeness, we should probably avoid it according to our current coding standard ;)).

Reply:
Fixed, removed unnamed structs/unions.